plugins/common/adx/adx_commons.go

```diff
@@ -0,0 +1,286 @@
+package adx_commons
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-kusto-go/kusto"
+	kustoerrors "github.com/Azure/azure-kusto-go/kusto/data/errors"
+
+	"github.com/Azure/azure-kusto-go/kusto/ingest"
+	"github.com/Azure/azure-kusto-go/kusto/kql"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/serializers/json"
+)
+
+var sampleConfig string
+
+type AzureDataExplorer struct {
+	Endpoint        string          `toml:"endpoint_url"`
```

Comment:
should this be endpoint_url ? or should it be cluster_url for consistency? We can check what it was earlier? , Is it ingest or cluster URL?

Reply:
Leaving it for now. Will take it in next change